tests/utils/console_test.py

```diff
@@ -0,0 +1,162 @@
+    def test_println_isquiet_and_randomized_docker_or_istty(self, patched_print, patched_isatty):
+        console.QUIET = True
+        random_boolean = random.choice([True, False])
+        patched_isatty.return_value, console.RALLY_RUNNING_IN_DOCKER = random_boolean, not random_boolean
+
+        console.println(msg="Unittest message")
+        patched_print.assert_not_called()
+
+
+# pytest style class names need to start with Test and don't need to subclass
+class TestCmdLineProgressReporter:
+    @mock.patch("sys.stdout.flush")
+    @mock.patch("sys.stdout.isatty")
+    # monkey patching builtins might break pytest's internals
+    # (https://docs.pytest.org/en/latest/monkeypatch.html#global-patch-example-preventing-requests-from-remote-operations)
+    # but hopefully print doesn't fall in this category
+    @mock.patch("builtins.print")
+    @pytest.mark.parametrize("seed", range(20))
+    def test_print_when_isquiet_and_any_docker_or_istty(self, patched_print, patched_isatty, patched_flush, seed):
+        console.QUIET = True
+        random.seed(seed)
+        patched_isatty.return_value, console.RALLY_RUNNING_IN_DOCKER = random.choice([True, False]), random.choice([True, False])
+
+        message = "Unit test message"
+        width = random.randint(20, 140)
+        progress_reporter = console.CmdLineProgressReporter(width=width)
+        progress_reporter.print(message=message, progress=".")
+        patched_print.assert_not_called()
+
+
+    @mock.patch("sys.stdout.isatty")
+    @mock.patch("builtins.print")
+    def test_prints_when_isnotquiet_and_nodocker_and_isnotty(self, patched_print, patched_isatty):
+        console.QUIET = False
+        patched_isatty.return_value, console.RALLY_RUNNING_IN_DOCKER = False, False
+
+        message = "Unit test message"
+        width = random.randint(20, 140)
+        progress_reporter = console.CmdLineProgressReporter(width=width)
+        progress_reporter.print(message=message, progress=".")
+        patched_print.assert_not_called()
+
+    @mock.patch("sys.stdout.flush")
+    @mock.patch("sys.stdout.isatty")
+    @mock.patch("builtins.print")
+    @pytest.mark.parametrize("seed", range(20))
+    def test_prints_when_isnotquiet_and_randomized_docker_or_istty(self, patched_print, patched_isatty, patched_flush, seed):
+        console.QUIET = False
+        random.seed(seed)
+        random_boolean = random.choice([True, False])
+        patched_isatty.return_value, console.RALLY_RUNNING_IN_DOCKER = random_boolean, not random_boolean
+
+        message = "Unit test message"
+        width = random.randint(20, 140)
+        progress_reporter = console.CmdLineProgressReporter(width=width)
+        progress_reporter.print(message=message, progress=".")
+        patched_print.assert_has_calls([
+            mock.call(" " * width, end=""),
+            mock.call("\x1b[{}D{}{}.".format(width, message, " "*(width-len(message)-1)), end="")
+        ])
+
+    @mock.patch("sys.stdout.flush")
```

Comment:
There should be no need to mock `flush` because it is not called. Or are you doing this for consistency? Should we then check that `flush` is not called?